aten/src/ATen/functorch/BatchRulesScatterOps.cpp

```diff
@@ -1056,6 +1056,132 @@ std::tuple<Tensor,optional<int64_t>> masked_fill_scalar_batch_rule(
   return std::make_tuple(result, 0);
 }
 
+std::tuple<Tensor,optional<int64_t>> index_fill_int_scalar_batch_rule_impl(
+    Tensor & self, optional<int64_t> self_bdim,
+    int dim,
+    const Tensor & index, optional<int64_t> index_bdim,
+    const Scalar & value,
+    const bool inplace) {
+
+    const auto self_logical_rank = rankWithoutBatchDim(self, self_bdim);
+    Tensor self_ = moveBatchDimToFront(self, self_bdim);
+    Tensor index_ = moveBatchDimToFront(index, index_bdim);
+    dim = maybe_wrap_dim(dim, self_logical_rank);
+
+    if (inplace && !self_bdim.has_value()) {
+      vmapIncompatibleInplaceError("index_fill_");
+    }
+
+    if (!index_bdim) {
+      self_.unsqueeze_(-1);
+      self_.index_fill_(dim + 1, index_, value);
+      self_.squeeze_(-1);
```

Comment:
To check my understanding: the `unsqueeze_(-1)` and the `squeeze_(-1)` is to handle `self_` tensors that have shape `[]`?

Reply:
Yes, I can make it specific for this case.

Reply:
That would be nice, it saves us some calls, especially because this case is not that common